pkg/controllers/nodeclass/status/ami_test.go

```diff
@@ -311,6 +311,63 @@ var _ = Describe("NodeClass AMI Status Controller", func() {
 		))
 		Expect(nodeClass.StatusConditions().IsTrue(v1.ConditionTypeAMIsReady)).To(BeTrue())
```

Comment:
Should we add vaildation if the reason and message for the status condation is not set of non-deprecated AMIs?

Reply:
Addressed, added an additional test